### PR TITLE
Recycler will produce npe error when multiple recycled at different thread

### PR DESCRIPTION
**Pull request by zhaojigang**
fix https://github.com/netty/netty/issues/8220

### CODE REVIEW

File: common/src/test/java/io/netty/util/RecyclerTest.java [hunk 1]
```diff
@@ -81,6 +81,38 @@ public void testMultipleRecycle() {
...
+        } catch (InterruptedException e) {
+        }
+
+        final Thread thread2 = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    object.recycle();
+                } catch (IllegalStateException e) {
+                }
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/RecyclerTest.java]: store the `IllegalStateException` in an `AtomicReference` and assert it after `thread2.join()` returns.

File: common/src/test/java/io/netty/util/RecyclerTest.java [hunk 2]
```diff
@@ -81,6 +81,38 @@ public void testMultipleRecycle() {
...
+        final Thread thread1 = new Thread(new Runnable() {
+            @Override
+            public void run() {
+                object.recycle();
+            }
+        });
+        thread1.start();
+        try {
+            thread1.join();
+        } catch (InterruptedException e) {
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/RecyclerTest.java]: no need to catch this... just let it throw.

File: common/src/test/java/io/netty/util/RecyclerTest.java [hunk 3]
```diff
@@ -81,6 +81,38 @@ public void testMultipleRecycle() {
...
+                    object.recycle();
+                } catch (IllegalStateException e) {
+                }
+            }
+        });
+        thread2.start();
+        try {
+            thread2.join();
+        } catch (InterruptedException e) {
+        }
```
Comment (normanmaurer) [inline common/src/test/java/io/netty/util/RecyclerTest.java]: no need to catch this... just let it throw.

**Comment by normanmaurer** [comment]
Also please fix the check style error : https://garage.netty.io/teamcity/viewLog.html?buildId=32568&buildTypeId=pulls_netty&guest=1

**Comment by zhaojigang** [comment]
ok, thanks!!! I will do it now!

**Pull request closed without merging**